<?xml version="1.0"?>
<launch>
    <arg name="mav_name" default="firefly"/>
    <arg name="frame_id" default="odom" />
    <arg name="voxel_size" default="0.2" />
    <arg name="clear_radius" default="1.0" />

    <group ns="$(arg mav_name)">
        <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr" clear_params="true">
            <remap from="pointcloud" to="vi_sensor/camera_depth/depth/points"/>
            <remap from="voxblox_node/esdf_map_out" to="esdf_map" />
            <remap from="voxblox_node/tsdf_map_out" to="tsdf_map" />
      
            <param name="publish_tsdf_map" value="true" />
            <param name="publish_esdf_map" value="true" />
            <param name="update_mesh_every_n_sec" value="0.25" />
            <param name="update_esdf_every_n_sec" value="0.25" />
            <param name="publish_map_every_n_sec" value="0.1" />
      
            <param name="method" value="fast" />
            <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
            <param name="tsdf_voxels_per_side" value="16" />
            <param name="esdf_max_distance_m" value="2.0" />
            <param name="max_ray_length_m" value="10.0" />
            <param name="voxel_carving_enabled" value="true" />
            <param name="color_mode" value="color" />
            <param name="use_tf_transforms" value="true" />
            <param name="min_time_between_msgs_sec" value="0.10" />
            <param name="clear_sphere_for_planning" value="true" />
            <param name="occupied_sphere_radius" value="4.0" />
            <param name="clear_sphere_radius" value="$(arg clear_radius)" />
            <param name="slice_level" value="1.0" />
            <param name="world_frame" value="$(arg frame_id)" />
            <param name="verbose" value="false" />
      
            <param name="esdf_add_occupied_crust" value="true" />
            <param name="publish_pointclouds" value="true" />
            <param name="publish_slices" value="true" />
            <param name="esdf_min_diff_m" value="0.0" />
            <param name="publish_traversable" value="true" />
        </node>
    </group>
</launch>